BlueprintUI/Sources/Layout/ConstrainedSizeFraction.swift

```diff
@@ -0,0 +1,94 @@
+    ///   the width is unconstrained. By default, `nil`.
+    ///   - height: The fraction of the parent's vertical layout space the content element should occupy. `nil` if
+    ///   the height is unconstrained. By default, `nil`.
+    ///   - wrapping: The content element.
+    public init(width: CGFloat? = nil, height: CGFloat? = nil, wrapping wrappedElement: Element) {
+        precondition(
+            width.map { $0 >= 0 && $0 <= 1} ?? true,
+            "The provided width fraction must be a value in the range of `0...1`."
+        )
+        precondition(
+            height.map { $0 >= 0 && $0 <= 1} ?? true,
+            "The provided height fraction must be a value in the range of `0...1`."
+        )
+
+        self.width = width
+        self.height = height
+        self.wrappedElement = wrappedElement
+    }
+
+    public var content: ElementContent {
+        ElementContent(
+            child: wrappedElement,
+            layout: Layout(width: width, height: height)
+        )
+    }
+
+    public func backingViewDescription(bounds: CGRect, subtreeExtent: CGRect?) -> ViewDescription? {
+        return nil
+    }
+
+    private struct Layout: SingleChildLayout {
+        var width: CGFloat?
+        var height: CGFloat?
+
+        func measure(in sizeConstraint: SizeConstraint, child: Measurable) -> CGSize {
+            /// Get the target layout space, constraining it by the specified fractions.
+            let targetWidth = width.map { $0 * sizeConstraint.width.maximum }
+            let targetHeight = height.map { $0 * sizeConstraint.height.maximum }
+
+            let constraint = SizeConstraint(
+                width: targetWidth.map { .atMost($0) } ?? .unconstrained,
+                height: targetHeight.map { .atMost($0) } ?? .unconstrained
+            )
+
+            /// Measure the child in the constrained layout space.
+            let measurement = child.measure(in: constraint)
+
+            /// Override the child's measurement with a target dimension if one exists.
+            return CGSize(
+                width: targetWidth ?? measurement.width,
+                height: targetHeight ?? measurement.height
+            )
+        }
+
+        func layout(size: CGSize, child: Measurable) -> LayoutAttributes {
+            return LayoutAttributes(size: size)
+        }
+    }
+}
+
+
+public extension Element {
+    /// Constrains the element to a fraction of its parent's size.
+    ///
+    /// - parameters:
+    ///   - width: The fraction of the parent's horizontal layout space the element should occupy. `nil`
+    ///   if the width is unconstrained. By default, `nil`.
+    ///   - height: The fraction of the parent's vertical layout space the element should occupy. `nil`
+    ///   if the height is unconstrained. By default, `nil`.
+    func constrainedTo(
+        widthFraction width: CGFloat? = nil,
```

Comment:
Naming nit: what do you think about `fractionalWidth` / `fractionalHeight`? could also use for the properties of the element themselves, it's not immediately obvious that setting the `element.width` is setting a fractional width

Reply:
I don't mind the update here, and considered it. If anyone wants to tiebreak, this I'd welcome that, and if not, I'll just change it.

If we update the property name, we're adding a bit of a stutter `ConstrainedSizeFraction.widthFraction`, which I'd prefer to avoid. Again, not strongly opposed, so please let me know if you'd like me to make this change.

Reply:
Could all of the properties be updated to be `fractional*`, including the vars and init?

Reply:
Yep, that's the tiebreak! I will update now.